src/App.jsx

```diff
@@ -14,33 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
-
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
-
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
-
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
-
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+const SortType = {
+  NONE: 'NONE',
+  ALPHABET: 'ALPHABET',
+  LENGTH: 'LENGTH',
+};
+
+function getReorderedGoods(goods, { sortType, isReversed }) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  } else if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isReversed: false,
+      sortType: SortType.NONE,
```

Comment:
The initial state for `sortType` is set to `SortType.NONE`. However, the `getReorderedGoods` function does not handle the `NONE` sort type explicitly. Consider adding a condition in `getReorderedGoods` to handle `SortType.NONE` or ensure that the UI reflects the unsorted state when `NONE` is selected.